src/Database/LSMTree/Model/Normal.hs

```diff
@@ -51,10 +53,28 @@ class SomeSerialisationConstraint a where
     -- Note: cannot fail.
     deserialise :: BS.ByteString -> a
 
+-- Note: we might add extra requirements,
+-- as this instance may produce arbitrary large serialised values.
+-- Do we need to limit maximum serialised size?
+-- Or even require it to be constant (and known in advance?)
```

Comment:
Yes, this is a good point, but we expect the key size constraints to be part of the table config, not global for the key type. In particular, we expect to have table keys be a fixed size, but with the fixed size different for different tables, and values to have some maximum size (though implementation defined not per-table).

Reply:
Isn't that what's kind of what `newtype`s are for?